internal/querynode/shard_cluster.go

```diff
@@ -611,15 +612,23 @@ func (sc *ShardCluster) HandoffSegments(info *querypb.SegmentChangeInfo) error {
 
 // appendHandoff adds the change info into pending list and returns the token.
 func (sc *ShardCluster) applySegmentChange(info *querypb.SegmentChangeInfo, onlineSegmentIDs []UniqueID) int64 {
+
+	// the suspects growing segment ids
+	// first all online segment shall be tried, for flush-handoff only puts segment in onlineSegments
+	// and we need to try all offlineSegments in case flush-compact-handoff case
+	possibleGrowingToRemove := make([]UniqueID, 0, len(info.OfflineSegments)+len(onlineSegmentIDs))
+	offlineSegments := typeutil.UniqueSet{} // map stores offline segment id for quick check
+	for _, offline := range info.OfflineSegments {
+		offlineSegments.Insert(offline.GetSegmentID())
+		possibleGrowingToRemove = append(possibleGrowingToRemove, offline.GetSegmentID())
+	}
+	// add online segment ids to suspect list
+	possibleGrowingToRemove = append(possibleGrowingToRemove, onlineSegmentIDs...)
```

Comment:
onlineSegmentIDs should be inserted into offlineSegments?
offlineSegments.Insert(onlineSegmentIDs...)